### PR TITLE
reporter/github: checking the response status code

### PR DESCRIPTION
**Pull request by iblis17**
- and show the github response if error

----
TODO

- [x] unit test

### CODE REVIEW

File: master/buildbot/reporters/github.py [hunk 1]
```diff
@@ -171,19 +171,24 @@ def send(self, build):
                     issue=issue,
                     description=description
                 )
+
+                assert response.code // 100 == 2
```
Comment (tardyp) [inline master/buildbot/reporters/github.py]: Please do not use assert. assert is supposed to be disabled in production.

Also // 100 == 2 is a bit hard to read

Comment (iblis17) [reply master/buildbot/reporters/github.py]: Which method do you prefer to check the status code? (I personally always use `//`

Comment (tardyp) [reply master/buildbot/reporters/github.py]: I would create a helper method isStatus2XX(response)

Comment (iblis17) [reply master/buildbot/reporters/github.py]: ha, okay.

**Comment by iblis17** [comment]
I added some test cases.

**Pull request merged**